WordPress/src/main/java/org/wordpress/android/util/UploadWorker.kt

```diff
@@ -0,0 +1,88 @@
+package org.wordpress.android.util
+
+import android.content.Context
+import androidx.work.Constraints
+import androidx.work.ExistingPeriodicWorkPolicy
+import androidx.work.ExistingWorkPolicy
+import androidx.work.ListenableWorker
+import androidx.work.NetworkType
+import androidx.work.OneTimeWorkRequestBuilder
+import androidx.work.Operation
+import androidx.work.PeriodicWorkRequestBuilder
+import androidx.work.WorkManager
+import androidx.work.WorkRequest
+import androidx.work.Worker
+import androidx.work.WorkerFactory
+import androidx.work.WorkerParameters
+import androidx.work.workDataOf
+import kotlinx.coroutines.runBlocking
+import org.wordpress.android.WordPress
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.store.SiteStore
+import org.wordpress.android.ui.uploads.LocalDraftUploadStarter
+import java.util.concurrent.TimeUnit.HOURS
+
+class UploadWorker(
+    appContext: Context,
+    workerParams: WorkerParameters,
+    private val localDraftUploadStarter: LocalDraftUploadStarter,
+    private val siteStore: SiteStore
+) : Worker(appContext, workerParams) {
+    companion object {
+        private const val UPLOAD_FROM_ALL_SITES = -1
+    }
+
+    override fun doWork(): Result {
+        runBlocking {
+            val job = when (val localSiteId = inputData.getInt(WordPress.LOCAL_SITE_ID, UPLOAD_FROM_ALL_SITES)) {
+                UPLOAD_FROM_ALL_SITES -> localDraftUploadStarter.queueUploadFromAllSites()
+                else -> siteStore.getSiteByLocalId(localSiteId)?.let { localDraftUploadStarter.queueUploadFromSite(it) }
```

Comment:
I'm sorry I didn't see this the first time around. We can make this `UploadWorker` simpler by supporting `siteId` in `LocalDraftUploadStarter` instead. 

```kt
fun queueUploadFromSite(siteId: Long) {
```

With that, we don't need to have `SiteStore` as a dependency for `UploadWorker`. 

It's not terribly important though. It's up to you if you want to change it.

Reply:
IMO it's better to have the SiteModel early and since we need injection for the LocalDraftUploadStarter, it's not a big deal to keep the SiteStore here.

Reply:
Yeah, that's fine. 🙂